File: comparisons/MinsC2Rust/c_algorithm/Output/test_project/src/compare_int.rs

pub fn int_compare<T: PartialOrd>(location1: &T, location2: &T) -> i32 {
    if location1 < location2 {
        -1
    } else if location1 > location2 {
        1
    } else {
        0
    }
}

pub fn int_equal<T: PartialEq>(vlocation1: &T, vlocation2: &T) -> bool {
    vlocation1 == vlocation2
}

File: c-algorithm/src/compare-int.c
// 识别到的包含指令
#include "compare-int.h"

// 识别到的函数定义
// 函数: int_equal (line 25)
int int_equal(void *vlocation1, void *vlocation2)
{
	int *location1;
	int *location2;

	location1 = (int *) vlocation1;
	location2 = (int *) vlocation2;

	return *location1 == *location2;
}

// 函数: int_compare (line 36)
int int_compare(void *vlocation1, void *vlocation2)
{
	int *location1;
	int *location2;

	location1 = (int *) vlocation1;
	location2 = (int *) vlocation2;

	if (*location1 < *location2) {
		return -1;
	} else if (*location1 > *location2) {
		return 1;
	} else {
		return 0;
	}
}
